app.py

```diff
@@ -15,9 +15,108 @@
+
+
+def push_thoth_sli_weekly_metric(
+    weekly_value_metric: Metric, prometheus_registry: CollectorRegistry, pushgateway_endpoint: str
+):
+    """Push Thoth SLI weekly metric to PushGateway."""
+    weekly_metric = Gauge("thoth_sli_weekly", "Weekly Thoth Service Level Indicators", registry=prometheus_registry)
+
+    weekly_value_metric = float(weekly_value_metric[0]["value"][1])
+    weekly_metric.set(weekly_value_metric)
+
+    push_to_gateway(pushgateway_endpoint, job="Weekly Thoth SLI", registry=prometheus_registry)
+
+    return weekly_value_metric
+
+
+def generate_email(learned_packages: int):
+    """General email to be sent."""
+    end_time = datetime.datetime.now()
+    start_time = end_time - datetime.timedelta(days=7)
+    start_time_epoch = int(start_time.timestamp() * 1000)
+    end_time_epoch = int(end_time.timestamp() * 1000)
+
+    # TODO: Show one single dashboard for all metrics to be shown.
+    grafana_dashboard_url = (
+        f"https://grafana.datahub.redhat.com/dashboard/db/thoth-knowledge-graph-content-metrics-stage?"
+        + f"refresh=1m&panelId=23&fullscreen&orgId=1&from={start_time_epoch}&to={end_time_epoch}"
+    )
+
+    return MIMEText(
+        f"<strong>Thoth Solved {learned_packages} Python Packages in the last week.</strong> \
+            <br><br> \
+            Python Packages Information collected in the last week by Thoth. \
+            <br> \
+            The following dashboard panel contains Python Packages info collected: <a href='{grafana_dashboard_url}'> Python Packages Info.</a> \
+            <br>",
+        "html",
+    )
+
+
+def send_sli_email(server: str, sender_address: str, recipients: str, learned_packages: int):
+    """Send email about Thoth Service Level Objectives."""
+    _MAIL_SERVER = smtplib.SMTP(server)
+
+    msg = MIMEMultipart()
+    msg["Subject"] = "Thoth Weekly Service Level Indicators"
+    msg["From"] = sender_address
+    msg["To"] = recipients
+
+    email_message = generate_email(int(learned_packages))
+
+    msg.attach(email_message)
+
+    _MAIL_SERVER.sendmail(sender_address, recipients, msg.as_string())
+
+
+def main():
+    """Main function for Thoth Service Level Objectives (SLO) Reporter."""
+    _SERVER = os.environ["SMTP_SERVER"]
+    _SENDER_ADDRESS = os.environ["SENDER_ADDRESS"]
+    _ADDRESS_RECIPIENTS = os.environ["EMAIL_RECIPIENTS"]
+
+    _THANOS_URL = os.environ["THANOS_ENDPOINT"]
+    _THANOS_TOKEN = os.environ["THANOS_ACCESS_TOKEN"]
+    _PUSHGATEWAY_ENDPOINT = os.environ["PROMETHEUS_PUSHGATEWAY_URL"]
+
+    prometheus_registry = CollectorRegistry()
+
+    # TODO: Collect all important metrics in a different function
+    pc = PrometheusConnect(url=_THANOS_URL, headers={"Authorization": f"bearer {_THANOS_TOKEN}"}, disable_ssl=True)
+    query = (
+        'thoth_graphdb_total_main_records{instance="metrics-exporter-thoth-frontend-stage.cloud.paas.psi.redhat.com:80", main_table="python_package_version"}'
```

Comment:
Can be the instance parametrized with an environment variable with this default value?